pkg/resource/k8s/configmapresource/new_config_map_to_update.go

```diff
@@ -23,6 +23,19 @@ func newConfigMapToUpdate(current, desired *corev1.ConfigMap) *corev1.ConfigMap
 	merged.Annotations = desired.Annotations
 	merged.Labels = desired.Labels
 
+	if allowedLabels != nil {
+		for k, v := range current.Labels {
+			if _, exist := desired.Labels[k]; exist {
+				// If label is already in desired spec, skip it.
+				continue
+			}
+
+			if _, ok := allowedLabels[k]; ok {
```

Comment:
I think we can make it simple, 
```suggestion
			if allowedLabels[k] {
```

Reply:
Thanks, I based this on the app resource but this is cleaner.